### ISSUE TITLE
[V3] Implement new Concept Exercise: time-duration

### ISSUE THREAD

**Issue by oanaOM** [exercism/go#1391]
This issue describes how to implement the `time-duration` concept exercise for the Go track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how concept time and duration is implemented in Go. 

## Learning objectives

- Define the relation between time.Time and time.Duration.
- Know how to measure time using time.Since ( e.g. execution time of a function -- no `defer` !) .
- Know how to use `time.Add` and `time.Sub`
- Know about time units and maybe do some calculations
- How to use time.Sleep (?)

## Out of scope

- Measuring time should to include `defer` 

## Concepts

- `time-duration`

## Prerequisites

- `time`

### Hints

- [Time package](https://golang.org/pkg/time/)

## Analyzer

This exercise does not require any specific logic to be added to the analyzer

## Implementing

To implement this exercise, please follow these [instructions](https://github.com/exercism/v3/blob/master/languages/go/reference/implementing-a-concept-exercise.md) and the below file structure.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by saurami** [comment]
@junedev I'd like to work on this

**Comment by saurami** [comment]
- [x] Finish exercise "Booking up for Beauty"
- [x] Review PR #2243 (and associated [Go By Example](https://gobyexample.com/time-formatting-parsing) page
- [x] Review concept exercise "[The Time Keeper](https://exercism.org/docs/building/tracks/stories/datetimes-time-keeper)"
- [] Create files and directories in `$PROJECT_ROOT/exercises/concept/the-time-keeper` based on [documentation](https://exercism.org/docs/building/tracks/concept-exercises)
- [] Solve and test exercise locally
- [] Create PR

**Comment by junedev** [comment]
@saurabhmshr Are you still working on this?

**Comment by saurabhmshr** [comment]
@junedev yes - I plan on having it out for review by Monday 24th October.

**Comment by IsaacG** [comment]
[Swift Scheduling](https://github.com/exercism/go/tree/main/exercises/practice/swift-scheduling) covers some of these boxes, but it's not in the concepts section

**Comment by thibault2705** [comment]
Hi @oanaOM, I created the [PR #3127 ](https://github.com/exercism/go/pull/3127) to add a new concept `time-duration` and a new concept exercise `time-keeper`. Let me know if the changes meet your expectations.
